transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java

```diff
@@ -245,6 +248,7 @@ public void put(K key, V value) throws CacheAccessException {
     } else {
       currentContext.addCommand(key, new StorePutCommand<V>(null, new XAValueHolder<V>(value, timeSource.getTimeMillis())));
     }
+    return PutStatus.PUT;
```

Comment:
If the current context already enlisted an Evict command, then this should be a NOOP. The same is true above BTW. `addCommand()` probably should return a boolean to indicate such case.